program/rust/src/accounts/score.rs

```diff
@@ -0,0 +1,158 @@
+        }
+
+        let symbol_index = self
+            .symbols
+            .iter()
+            .position(|&x| x == price_account)
+            .ok_or(ProgramError::InvalidArgument)?;
+
+        self.publisher_permissions[publisher_index][symbol_index] = true;
+        self.calculate_scores()
+    }
+
+    pub fn del_publisher(
+        &mut self,
+        publisher: Pubkey,
+        price_account: Pubkey,
+    ) -> Result<(), ProgramError> {
+        let publisher_index = self
+            .publishers
+            .iter()
+            .position(|&x| x == publisher)
+            .ok_or(ProgramError::InvalidArgument)?;
+
+        let symbol_index = self
+            .symbols
+            .iter()
+            .position(|&x| x == price_account)
+            .ok_or(ProgramError::InvalidArgument)?;
+
+        self.publisher_permissions[publisher_index][symbol_index] = false;
+        self.calculate_scores()
+    }
+
+    pub fn add_price(&mut self, symbol: Pubkey) -> Result<(), ProgramError> {
+        let symbol_index = self
+            .symbols
+            .iter()
+            .position(|&x| x == symbol)
+            .unwrap_or(self.num_symbols);
+
+        if symbol_index == self.num_symbols {
+            if self.num_symbols == PC_MAX_SYMBOLS as usize {
+                return Err(ProgramError::AccountDataTooSmall);
+            }
+
+            self.symbols[self.num_symbols] = symbol;
+            self.num_symbols += 1;
+        }
+
+        Ok(())
+    }
+
+    pub fn del_price(&mut self, symbol: Pubkey) -> Result<(), ProgramError> {
+        let symbol_index = self
+            .symbols
+            .iter()
+            .position(|&x| x == symbol)
+            .ok_or(ProgramError::InvalidArgument)?;
+
+        self.symbols[symbol_index] = self.symbols[self.num_symbols - 1];
+        self.symbols[self.num_symbols - 1] = Pubkey::default();
+        self.num_symbols -= 1;
+        self.calculate_scores()
+    }
+
+    pub fn calculate_scores(&mut self) -> Result<(), ProgramError> {
+        let symbol_scores: Vec<u32> = self
+            .symbols
+            .iter()
+            .enumerate()
+            .map(|(j, _)| {
+                let score = self
+                    .publisher_permissions
+                    .iter()
+                    .fold(0, |score, permissions| score + permissions[j] as u32);
+                max(score, self.z)
+            })
+            .collect();
+
+        for i in 0..self.num_publishers {
+            self.scores[i] = self
+                .symbols
+                .iter()
+                .enumerate()
+                .filter(|(j, _)| self.publisher_permissions[i][*j])
+                .map(|(j, _)| 1f64 / symbol_scores[j] as f64)
+                .sum();
+        }
+        Ok(())
+    }
+}
+
+impl PythAccount for PublisherScoresAccount {
+    const ACCOUNT_TYPE: u32 = PC_ACCTYPE_SCORE;
+    // Calculate the initial size of the account
+    const INITIAL_SIZE: u32 = 1000;
```

Comment:
this one doesn't seem right?

Reply:
it was a placeholder at the time. Fixed.